src/python_testing/mdns_discovery/mdns_async_service_info.py

```diff
@@ -15,121 +15,155 @@
+    def __init__(self,
+                 name: str | None = None,  # Fully qualified service name
+                 type_: str | None = None,  # Fully qualified service type name
+                 server: str | None = None  # Fully qualified name for service host (defaults to name)
+                 ) -> None:
+
+        # For AAAA address resolution, only server (hostname)
+        # is to be provided, this configuration is set by the
+        # AddressResolverIPv6 class which inherits from this class
+        if server and not (name and type_):
+            super().__init__(type_=server, name=server, server=server)
+            self._name = None
+            self.type = None
+            self.server = server
+            return
+
+        # Validate a fully qualified service name, instance or subtype
+        if not type_.endswith(service_type_name(name, strict=False)):
             raise BadTypeInNameException
 
+        super().__init__(type_=type_, name=name, server=server)
         self._name = name
         self.type = type_
-        self.key = name.lower()
+        self.server = server
 
     async def async_request(
         self,
         zc: Zeroconf,
         timeout: float,
-        question_type: Optional[DNSQuestionType] = None,
-        addr: Optional[str] = None,
+        question_type: DNSQuestionType | None = None,
+        addr: str | None = None,
         port: int = _MDNS_PORT,
-        record_type: DNSRecordType = None
     ) -> bool:
-        """Returns true if the service could be discovered on the
-        network, and updates this object with details discovered.
+        """Returns true if the service could be discovered on the network, and updates this
+        object with details discovered.
 
-        This method will be run in the event loop.
+        Custom override of `zeroconf.ServiceInfo.async_request` that prioritizes fresh data.
 
-        Passing addr and port is optional, and will default to the
-        mDNS multicast address and port. This is useful for directing
-        requests to a specific host that may be able to respond across
-        subnets.
+        Key differences from the base implementation:
+        - Bypasses known-answer caching to force a network query on every call.
+        - Clears the internal cache before querying to prevent stale results.
+        - Allows filtering by specific DNS record types (SRV, TXT, A, AAAA, etc.).
         """
         if not zc.started:
             await zc.async_wait_for_start()
+        if TYPE_CHECKING:
+            assert zc.loop is not None
+
+        # Force fresh queries by clearing cache and question history
+        zc.cache.cache.clear()
+        zc.question_history.clear()
+        self.async_clear_cache()
 
         now = current_time_millis()
 
-        if TYPE_CHECKING:
-            assert zc.loop is not None
+        # Absolute cutoff time after which the request stops if incomplete
+        deadline_ms = now + timeout
```

Comment:
here I am unsure: if seeing `float` I would have assumed timeout is in seconds. But now I see this is in ms ... please add time units to parameters to avoid errors.
It seems this should be `now_ms` and `timeout_ms` (or deadline should be deadline_sec)